Samples/DirectMLNpuInference/main.cpp

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include <dxcore_interface.h>
+#include <dxcore.h>
+
+#include "onnxruntime_cxx_api.h"
+#include "dml_provider_factory.h"
+
+#include "TensorHelper.h"
+
+using Microsoft::WRL::ComPtr;
+
+void InitializeDirectML(ID3D12Device1** d3dDeviceOut, ID3D12CommandQueue** commandQueueOut, IDMLDevice** dmlDeviceOut) {
```

Comment:
>InitializeDirectML( [](http://example.com/codeflow?start=5&length=19)

Should we just pass in a level required to this function so that it is more reusable?

Reply:
level?

Reply:
DXCORE_ADAPTER_ATTRIBUTE_D3D12_CORE_COMPUTE, D3D_FEATURE_LEVEL_11_0, D3D_FEATURE_LEVEL_12..., ect...

Reply:
this was shared in an earlier devkit